### PR TITLE
Added cleanup method for the Email Tracking DAO

### PR DESCRIPTION
**Pull request by troian88**
Added cleanup method for Email Tracking DAO

**Additional context**
https://github.com/BroadleafCommerce/QA/issues/4865

### CODE REVIEW

File: common/src/main/java/org/broadleafcommerce/common/email/dao/EmailReportingDaoImpl.java [hunk 1]
```diff
@@ -61,6 +69,25 @@ public EmailTarget createTarget() {
...
+        CriteriaBuilder builder = em.getCriteriaBuilder();
+        CriteriaQuery<EmailTracking> criteria = builder.createQuery(EmailTracking.class);
+        Root<EmailTrackingImpl> EmailTrackingRoot = criteria.from(EmailTrackingImpl.class);
+
+        criteria.select(EmailTrackingRoot);
+
+        Query query = em.createQuery(criteria);
+        query.setHint(QueryHints.HINT_CACHEABLE, true);
+
+        for (Object emailTracking : query.getResultList()) {
```
Comment (oleksiimiroshnyk) [inline common/src/main/java/org/broadleafcommerce/common/email/dao/EmailReportingDaoImpl.java]: I think potentially we can have large number of records here, so this is not optimal
I think we can do this via JPQL ?
https://vladmihalcea.com/bulk-update-delete-jpa-hibernate/

**Review by oleksiimiroshnyk — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/broadleafcommerce/common/email/dao/EmailReportingDaoImpl.java (hunk 1 above)
Comment (troian88) [reply common/src/main/java/org/broadleafcommerce/common/email/dao/EmailReportingDaoImpl.java]: updated

File: common/src/main/java/org/broadleafcommerce/common/email/dao/EmailReportingDaoImpl.java [hunk 2]
```diff
@@ -61,6 +66,14 @@ public EmailTarget createTarget() {
         return target;
     }
 
+    @Override
+    public void clearAllRecordsOlderThan(LocalDateTime date) {
+        int deleteCount = em.createQuery("DELETE FROM BLC_EMAIL_TRACKING WHERE DATE_SENT <= :date")
+            .setParameter("date", Timestamp.valueOf(date))
+            .executeUpdate();
+        LOG.debug("clearAllRecordsOlderThan: " + deleteCount + " items was removed.");
```
Comment (oleksiimiroshnyk) [inline common/src/main/java/org/broadleafcommerce/common/email/dao/EmailReportingDaoImpl.java]: I would probably make this more informative: deleted X records from BLC_EMAIL_TRACKING with date_sent less than Y ?
and maybe level info - we don't expect it to be executed too often, but can be debug, not sure.

**Review by oleksiimiroshnyk — APPROVED** [review]

**Comment by danielcolgrove** [comment]
@troian88 @oleksiimiroshnyk The blc_email_tracking table does not have an index on date_sent...to prevent a table scan should we add one?

**Review by danielcolgrove — APPROVED** [review]

**Review by riteshadhikari17 — APPROVED** [review]

**Pull request merged**